src/core/border_router/routing_manager.cpp

```diff
@@ -645,51 +743,79 @@ void RoutingManager::DeprecateOnLinkPrefix(void)
 }
 
 #if OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE
-void RoutingManager::EvaluateNat64Prefix(void)
+void RoutingManager::EvaluateNat64Prefix(const Ip6::Prefix &aNewInfraIfNat64Prefix)
```

Comment:
Can this function be simplified if we have a `Ip6::Prefix *mAdvertisedNat64Prefix` member points to the NAT64 prefix which is being advertised in Network Data (`null` if currently not advertising a NAT64 prefix)?

Reply:
Discussed offline. Using`*mAdvertisedNat64Prefix` to replace `mNat64PrefixStatus` might save some code, but the main logic in this function would not be simplified. We still need to compare `mInfraIfNat64Prefix`/`mLocalNat64Prefix` with the `preferredNat64PrefixConfig`/`aNewInfraIfNat64Prefix` in different conditions to make sure the expected prefix is advertised/withdrew.